integration_tests/src/main/python/array_test.py

```diff
@@ -80,4 +80,22 @@ def test_orderby_array_of_structs(data_gen):
     assert_gpu_and_cpu_are_equal_sql(
         lambda spark : unary_op_df(spark, data_gen),
         'array_table',
-        'select array_table.a, array_table.a[0].child0 as first_val from array_table order by first_val')
\ No newline at end of file
+        'select array_table.a, array_table.a[0].child0 as first_val from array_table order by first_val')
+
+
+@pytest.mark.parametrize('data_gen', [ArrayGen(sub_gen) for sub_gen in [byte_gen, short_gen, int_gen, long_gen,
+                                                                        FloatGen(no_nans=True), DoubleGen(no_nans=True),
+                                                                        string_gen, boolean_gen, date_gen, timestamp_gen]], ids=idfn)
+def test_array_contains(data_gen):
+    assert_gpu_and_cpu_are_equal_sql(
```

Comment:
We can generate scalar values. 

https://github.com/NVIDIA/spark-rapids/blob/d706e3ec92882adc5c05d94ba14b1fab9178d837/integration_tests/src/main/python/data_gen.py#L616-L637

Might need to switch from SQL to a dataframe though, but it should not be too bad.

```
@pytest.mark.parametrize('data_gen', byte_gen, short_gen, int_gen, long_gen,
                                                                        FloatGen(no_nans=True), DoubleGen(no_nans=True),
                                                                        string_gen, boolean_gen, date_gen, timestamp_gen], ids=idfn)
def test_array_contains(data_gen):
  arr_gen = ArrayGen(data_gen)
  lit = gen_scalar(data_gen)
  assert_gpu_and_cpu_are_equal(lambda spark: two_col_df(spark, arr_gen, data_gen).select(array_contains(col('a'), lit), array_contains(col('a'), col('b')), ...)
```